osquery/hashing/hashing.cpp

```diff
@@ -84,14 +93,10 @@ std::string Hash::digest() {
     }
     return digest.str();
   } else if (encoding_ == HASH_ENCODING_TYPE_BASE64) {
-    std::stringstream digest;
-    for (size_t i = 0; i < length_; i++) {
-      digest << hash[i];
```

Comment:
As it seems, `operator<<` should append `unsigned char` to the stream as a `char`. So the whole iteration can be simplified with just a copy.
The copy itself can be avoided, but this will require a few casts that look bad.

Looks like I missed a test for this case, will add.

Reply:
I've checked new tests with `HASH_ENCODING_TYPE_BASE64` with and wthout this particular change and don't see any difference in results.